src/v/cluster/rm_stm.cc

```diff
@@ -1640,7 +1644,7 @@ ss::future<tx::errc> rm_stm::abort_all_txes() {
 
     co_await ss::max_concurrent_for_each(
       _active_tx_producers, 5, [this, &last_err](const auto& producer) {
-          return mark_expired(producer.id()).then([&last_err](tx::errc res) {
+          return do_mark_expired(producer.id()).then([&last_err](tx::errc res) {
```

Comment:
is this commit fixing an existing bug?

The fact that this method has no synchronization seems a bit concerning at first glance (I was hoping it takes an exclusive state lock) and the behavior is further relaxed by this commit. AFAICT there is some external synchronization as this method is only called under a partition level writes disabled lock but wondering if the method here should grab an exlcusive lock too?

Reply:
Yes, you are right, I meant to hold exclusive lock here. Thank you!